src/main/java/com/hbm/tileentity/machine/TileEntityMachineStrandCaster.java

```diff
@@ -0,0 +1,333 @@
+                    return water.getFill() >= getWaterRequired() && steam.getFill() < steam.getMaxFill();
+
+            }
+        }
+
+        return false;
+    }
+
+    public DirPos[] getFluidConPos() {
+
+        ForgeDirection dir = ForgeDirection.getOrientation(this.getBlockMetadata() - BlockDummyable.offset);
+        ForgeDirection rot = dir.getRotation(ForgeDirection.UP);
+
+        return new DirPos[]{
+                new DirPos(xCoord + rot.offsetX * 2 - dir.offsetX, yCoord, zCoord + rot.offsetZ * 2 - dir.offsetZ, rot),
+                new DirPos(xCoord - rot.offsetX - dir.offsetX, yCoord, zCoord - rot.offsetZ - dir.offsetZ, rot.getOpposite()),
+                new DirPos(xCoord + rot.offsetX * 2 - dir.offsetX * 5, yCoord, zCoord + rot.offsetZ * 2 - dir.offsetZ * 5, rot),
+                new DirPos(xCoord - rot.offsetX - dir.offsetX * 5, yCoord, zCoord - rot.offsetZ + dir.offsetZ * 5, rot.getOpposite()),
+        };
+    }
+
+    public int[][] getMetalPourPos() {
+
+        ForgeDirection dir = ForgeDirection.getOrientation(this.getBlockMetadata() - BlockDummyable.offset);
+        ForgeDirection rot = dir.getRotation(ForgeDirection.UP);
+
+        return new int[][]{
+                new int[]{xCoord + rot.offsetX - dir.offsetX, yCoord + 2, zCoord + rot.offsetZ - dir.offsetZ},
+                new int[]{xCoord - dir.offsetX, yCoord + 2, zCoord - dir.offsetZ},
+                new int[]{xCoord + rot.offsetX, yCoord + 2, zCoord + rot.offsetZ},
+                new int[]{xCoord, yCoord + 2, zCoord},
+        };
+    }
+
+    @Override
+    public ItemMold.Mold getInstalledMold() {
+        if (slots[0] == null) return null;
+
+        if (slots[0].getItem() == ModItems.mold) {
+            return ((ItemMold) slots[0].getItem()).getMold(slots[0]);
+        }
+
+        return null;
+    }
+
+    @Override
+    public int getMoldSize() {
+        return getInstalledMold().size;
+    }
+
+    @Override
+    public boolean canAcceptPartialPour(World world, int x, int y, int z, double dX, double dY, double dZ, ForgeDirection side, Mats.MaterialStack stack) {
+
+        if (side != ForgeDirection.UP) return false;
+        for (int[] pos : getMetalPourPos()) {
+            if (pos[0] == x && pos[1] == y && pos[2] == z) {
+                return this.standardCheck(world, x, y, z, side, stack);
+            }
+        }
+        return false;
+
+    }
+
+    @Override
+    public boolean standardCheck(World world, int x, int y, int z, ForgeDirection side, Mats.MaterialStack stack) {
+        if (this.type != null && this.type != stack.material) return false;
+        return !(this.amount >= this.getCapacity() || getInstalledMold() == null);
+    }
+
+    @Override
+    public int getCapacity() {
+        ItemMold.Mold mold = this.getInstalledMold();
+        return mold == null ? 50000 : mold.getCost() * 10;
+    }
+
+    private int getWaterRequired() {
+        return getInstalledMold() != null ? 5 * getInstalledMold().getCost() : 50;
```

Comment:
The second half of the ternary assumes `getInstalledMold()` to be null, then calls a method anyway

Reply:
Given the water required only gets deducted if the thing can process, which in the case, it couldn't, this wouldn't actually affect anything
This is more of a failsafe

Reply:
I'd rather if your failsafe didn't include a built-in crash